.\"t
.\" Automatically generated by Pandoc 1.18
.\"
.TH "cancelTask" "1" "" "SequoiaDB User Manuals" ""
.hy
.SH NAME
.PP
cancelTask \- cancel the background task
.SH SYNOPSIS
.PP
\f[B]db.cancelTask(<id>, [isAsync])\f[]
.SH CATEGORY
.PP
Sdb
.SH DESCRIPTION
.PP
This function is used to cancel the background task.
.SH PARAMETERS
.IP \[bu] 2
id ( \f[I]number, required\f[] )
.RS 2
.PP
Task ID
.RE
.IP \[bu] 2
isAsync ( \f[I]boolean, optional\f[] )
.RS 2
.PP
Whether to cancel the task asynchronously, the default value is false, which
means that the task is canceled synchronously.
.RE
.SH RETURN VALUE
.PP
When the function executes successfully, there is no return value.
.PP
When the function fails, an exception will be thrown and an error message will
be printed.
.SH ERRORS
.PP
The common exceptions of \f[C]cancelTask()\f[] function are as follows:
.PP
.TS
tab(@);
l l l l l.
T{
0
T}@T{
Error Code
T}@T{
Error Type
T}@T{
Description
T}@T{
Solution
T}
_
T{
1
T}@T{
\-173
T}@T{
SDB_CAT_TASK_NOTFOUND
T}@T{
Task does not exist.
T}@T{
Use listTasks() to
T}
T{
T}@T{
T}@T{
T}@T{
T}@T{
check whether the task
T}
T{
T}@T{
T}@T{
T}@T{
T}@T{
exists.
T}
.TE
.PP
When the exception happens, use getLastErrMsg() to get the error message or use
getLastError() to get the error code.
For more details, refer to Troubleshooting.
.SH VERSION
.PP
v3.4 and above
.SH EXAMPLES
.PP
Cancel a split task asynchronously.
.IP
.nf
\f[C]
> var taskid1 = db.sample.employee.splitAsync("group1", "group2", 50)
> db.cancelTask(taskid1, true)
\f[]
.fi
